### PR TITLE
Add e2e test for authorization proxy server installation

### PR DESCRIPTION
**Pull request by alikdell**
# Description
Add e2e test for CSM Operator Supports Install of Authorization Proxy Server.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|  #https://github.com/dell/csm/issues/511 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by alikdell** [comment]
> Are these added as part of nightly tests? Also: Can we use the auth proxy server deployed in this test to work with the tests that deploy auth sidecar?

For now I am testing manually, I haven't look in nightly job yet. I am new to operator world, not sure we do multiple deployment in single scenario, I am looking into that

**Review by JacobGros — APPROVED** [review]
LGTM
would be a good idea to add this to nightly tests

**Comment by alikdell** [comment]
> LGTM would be a good idea to add this to nightly tests

Jenkins and enabling multiple resources apply need some more works, we take that as beyond FC

File: tests/e2e/testfiles/csm_authorization_proxy_server.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            value: "nginx" 
+
+          # Storage-service ingress configuration
+          # Default value: nginx
+          - name: "STORAGE_INGRESS_CLASSNAME"
+            value: "nginx" 
+
+          # Specify storage class for redis. Otherwise, default storage class is used
+          - name: "REDIS_STORAGE_CLASS"
+            value: "vxflexos"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/csm_authorization_proxy_server.yaml:77]: Does this storage class get created as apart of the test?

**Review by JacobGros — DISMISSED** [review]
still LGTM

File: tests/e2e/testfiles/values.yaml [hunk 2]
```diff
@@ -593,3 +587,12 @@
     - "Enable forceRemoveDriver on CR [1]"
     - "Delete custom resource [1]"
 
+- scenario: "Install Authorization Proxy Server"
+  path: "testfiles/csm_authorization_proxy_server.yaml"
+  modules:
+    - "authorizationproxyserver"
+  steps:
+    - "Given an environment with k8s or openshift, and CSM operator installed"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/values.yaml:595]: The proxy server doesn't support Openshift when deployed through the Operator

**Review by JacobGros — APPROVED** [review]
LGTM

File: tests/e2e/testfiles/csm_authorization_proxy_server.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            value: "nginx" 
+
+          # Storage-service ingress configuration
+          # Default value: nginx
+          - name: "STORAGE_INGRESS_CLASSNAME"
+            value: "nginx" 
+
+          # Specify storage class for redis. Otherwise, default storage class is used
+          - name: "REDIS_STORAGE_CLASS"
+            value: "local"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/csm_authorization_proxy_server.yaml:77]: Is this storage class automatically created or are you making this a prereq to running the E2E test?

Comment (alikdell) [reply tests/e2e/testfiles/csm_authorization_proxy_server.yaml:77]: This is pre-requisite

**Review by shaynafinocchiaro — APPROVED** [review]

**Comment by shaynafinocchiaro** [comment]
Will you be adding steps in the /tests/README.md for how to execute the test?

**Comment by alikdell** [comment]
>

**Comment by alikdell** [comment]
> Will you be adding steps in the /tests/README.md for how to execute the test?

Yes

**Pull request merged**